File: SIC_Backend/Repositories/IUsersRepository.cs
using SIC_Backend.Data.DTOs;
using SIC_Backend.Data.Models;

namespace SIC_Backend.Repositories
{
    public interface IUsersRepository
    {
        public UserDTO GetUserById(string id);
        public UserDTO GetUserByUsername(string username);
        public Task<bool> IsUserNameAvailableAsync(string username);
        public Task<IEnumerable<NotificationToken>> GetNotificationTokens(string userId);
    }
}

File: sic_home/lib/repositories/interfaces/i_users_repository.dart
import 'package:sic_home/models/authentication/notification_token.dart';
import 'package:sic_home/models/user.dart';

abstract class IUsersRepository {
  Future<User> getUserById(int id);

  Future<User> getUserByUserName(String userName);

  Future<bool> isUserNameAvailable(String userName);

  Future<List<NotificationToken>> getNotificationTokens(String userId);
}
